aiohttp/connector.py

```diff
@@ -1126,3 +1127,53 @@ def path(self) -> str:
             raise ClientConnectorError(req.connection_key, exc) from exc
 
         return cast(ResponseHandler, proto)
+
+
+class NamedPipeConnector(BaseConnector):
+    """Named pipe connector.
+    Only supported by the proactor event loop.
+    See also: https://docs.python.org/3.7/library/asyncio-eventloop.html
+
+    path - Windows named pipe path.
+    keepalive_timeout - (optional) Keep-alive timeout.
+    force_close - Set to True to force close and do reconnect
+        after each request (and between redirects).
+    limit - The total number of simultaneous connections.
+    limit_per_host - Number of simultaneous connections to one host.
+    loop - Optional event loop.
+    """
+
+    def __init__(self, path: str, force_close: bool=False,
+                 keepalive_timeout: Union[object, float, None]=sentinel,
+                 limit: int=100, limit_per_host: int=0,
+                 loop: Optional[asyncio.AbstractEventLoop]=None) -> None:
```

Comment:
```suggestion
                 loop: Optional[asyncio.ProactorEventLoop]=None) -> None:
```

Reply:
Changing this causes(while running `mypy`)
```
aiohttp\connector.py:1149: error: Invalid type "asyncio.ProactorEventLoop"
```

Reply:
Odd... are you running mypy under Python 3.7?

Reply:
Maybe we could add a conditional for this to work under Linux?

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    try:
        from asyncio import ProactorEventLoop
    except ImportError:
        from asyncio import AbstractEventLoop
        class ProactorEventLoop(ProactorEventLoop):
            """ProactorEventLoop stub for mypy."""
```

Reply:
I am infact running mypy under python 3.7.
Also I too tried with 
```
if TYPE_CHECKING:
    from asyncio import ProactorEventLoop
```
Still the same error.
I'll try again, maybe I did something wrong in rush

Reply:
Nope, still happening. It is odd

Reply:
It's because you're trying to make the import under Linux and docs say it's available only under windows

Reply:
It fails on my computer too, where I'm running it on windows 🤔

Reply:
attribute access is not static analysis by MyPy